src/App/MainPage.xaml

```diff
@@ -37,123 +44,128 @@
-            <Grid.ColumnDefinitions>
-                <ColumnDefinition Width="*"/>
-                <ColumnDefinition Width="*"/>
-            </Grid.ColumnDefinitions>
-
-            <StackPanel Grid.Column="1" Orientation="Horizontal" VerticalAlignment="Top" HorizontalAlignment="Right">
-            </StackPanel>
-        </Grid>
 
         <Grid x:Name="LayoutRoot" Width="Auto" Height="Auto" HorizontalAlignment="Stretch" Margin="30,40,30,0" VerticalAlignment="Stretch">
-            
             <Grid.RowDefinitions>
                 <RowDefinition Height="Auto" />
                 <RowDefinition Height="Auto" />
                 <RowDefinition Height="10" />
                 <RowDefinition Height="*" />
             </Grid.RowDefinitions>
+            
+            <Grid x:Name="HeaderGrid">
+                <Grid.ColumnDefinitions>
+                    <ColumnDefinition Width="*"></ColumnDefinition>
+                    <ColumnDefinition Width="*"></ColumnDefinition>
+                </Grid.ColumnDefinitions>
+                <Grid.RowDefinitions>
+                    <RowDefinition Height="Auto" />
+                    <RowDefinition Height="Auto" />
+                </Grid.RowDefinitions>
+                <TextBlock x:Name="Header" x:Uid="FactoryOrchestrator" Text="" TextWrapping="Wrap" VerticalAlignment="Center" HorizontalAlignment="Left" Grid.Column="0" FontWeight="Bold" Style="{StaticResource TitleTextBlockStyle}"/>
+                <StackPanel x:Name = "MainControlPanel" Orientation="Horizontal" HorizontalAlignment="Right" Padding="40,0,0,0" Grid.Row="0" Grid.Column="1">
+                    <StackPanel x:Name="IndicatorPanel" Orientation="Horizontal" Background="{ThemeResource ControlAltFillColorTransparent}" HorizontalAlignment="Left" CornerRadius="5" Padding="1">
+                        <StackPanel Orientation="Vertical" Visibility="Collapsed" VerticalAlignment="Center" HorizontalAlignment="Left" Margin="5">
+                            <TextBlock x:Name="NetworkName" Text="" Style="{StaticResource CaptionTextBlockStyle}" VerticalAlignment="Center" TextWrapping="Wrap" IsTextSelectionEnabled="True"/>
+                            <TextBlock x:Name="NetworkIp" Text="" Style="{StaticResource CaptionTextBlockStyle}" VerticalAlignment="Center" TextWrapping="Wrap" IsTextSelectionEnabled="True"/>
+                        </StackPanel>
+                        <StackPanel Orientation="Vertical" VerticalAlignment="Center" HorizontalAlignment="Left" Margin="5">
+                            <TextBlock x:Name="OEMVersionHeader" Text="" VerticalAlignment="Center" HorizontalAlignment="Left" TextWrapping="Wrap" Style="{StaticResource CaptionTextBlockStyle}" IsTextSelectionEnabled="True"/>
+                            <TextBlock x:Name="AppVersionHeader" Text=""  VerticalAlignment="Top" HorizontalAlignment="Left" TextWrapping="Wrap" Style="{StaticResource CaptionTextBlockStyle}" IsTextSelectionEnabled="True"/>
```

Comment:
Id also keep os version in the forefront, as OEMs frequently are juggling multiple OS images and devices might boot without working input/connectivity, so being able to quickly tell os version is useful.

Reply:
We have learned that app version is very useful for our own diagnosis as we can identify what is running in screen shots.

If there is no working input/connectivity, how is the information on OS version used?

Reply:
"Hey microsoft i'm running factoryos build blah and i dont see input" 😄. Also not every device can have network working. ie CENT had issues there early in bringup.

My 2c is that very few apps so prominently display their version and i feel like oems care more about network connectivity & os version > app version.

you could consider adding a footer with more info too? or rotating through info like i did with the networks, but that wont help screenshots

Reply:
I like the idea of a footer, with small print. It would be akin to having a watermark on the Windows desktop. That would serve our purposes for analyzing screenshots.

I'd like to be deliberate about the layout, now that we will have multiple people working on this, and potentially multiple customers. We need to be able to innovate on the design, but with context about exactly which content is required by customers to be constantly visible. Ideally we would have this as part of a design spec, with each required element called out with the scenario it supports. PM involvement would be helpful for this, so we can also solicit direct feedback.